Modules/_stat.c

```diff
@@ -295,9 +295,21 @@ _PyLong_AsMode_t(PyObject *op)
     unsigned long value;
     mode_t mode;
 
-    value = PyLong_AsUnsignedLong(op);
-    if ((value == (unsigned long)-1) && PyErr_Occurred())
+    if (PyLong_Check(op)) {
+        value = PyLong_AsUnsignedLong(op);
```

Comment:
This code path looks like a micro-optimization (to avoid PyNumber_Index). I don't think that it's worth it.

Reply:
Not only. Many functions like `PyLong_AsUnsignedLong()` and `PyLong_AsUnsignedLongMask()`, many setters and converter, only call `__index__()` if the argument not `int`. So there would be a subtle behavior difference when the argument is an instance of the `int` subclass which defines `__index__()`. And calling `__index__()` for `int` subclass adds overhead.

But this is not completely consistent, so this may not make large difference.